src/language-js/needs-parens.js

```diff
@@ -155,6 +159,23 @@ function needsParens(path, options) {
     return true;
   }
 
+  // `export default function` or `export default class` can't be followed by
+  // anything after. So an expression like `export default (function(){}).toString()`
+  // needs to be followed by a parentheses
+  if (parent.type === "ExportDefaultDeclaration") {
+    // IIFE already include parenthesis between the expression and invocation
+    // so we don't need to handle here.
```

Comment:
Would be good to have a test with IIFE.

Reply:
I was already adding those 👍 
Thanks for the suggestion